### PR TITLE
addScript

### PR DESCRIPTION
**Pull request by vatsa287**
Script for quick starting the kadalu operator.

### CODE REVIEW

**Comment by amarts** [comment]
@vatsa287 I am not sure such scripts are of a choice to many people because they may have some experience already by the time they start with deploying the project. May be it is a good thing to add in a blog, and say you did use it.

But thinking more about it, you can extend such behavior with `kubectl kadalu` CLI itself. If there are no arguments given, it can ask for next steps interactively. And as it already has options to manage yaml files, that may be a better place to start this.

**Review by amarts — CHANGES_REQUESTED** [review]
marking request changes as I prefer it to be done through CLI at least.

**Comment by vatsa287** [comment]
> marking request changes as I prefer it to be done through CLI at least.

sure thing.

**Comment by amarts** [comment]
IMO, this flow is not required for any k8s user, as most of them would be aware of how to setup the config files! Instead the best PR to work on would be the #45 as a first-timer!

**Comment by vatsa287** [comment]
Yes , I will look into it and try to learn

**Comment by amarts** [comment]
Can you make the script without `kubectl-kadalu` package? but with a template storage yaml file, which gets replaced with proper variables based on the arguments?

File: extras/scripts/kadalu-quick-start-generic.sh [hunk 1]
```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+#Deploy Kadalu Operator
+echo -e "Do you want to deploy Kadalu or is it already deployed [y/n]"
+read answer1
+if [ $answer1 == 'y' ] ; then 
+
+       sudo kubectl create -f https://kadalu.io/operator-latest.yaml
```
Comment (amarts) [inline extras/scripts/kadalu-quick-start-generic.sh:8]: sudo kubectl apply -f https://raw.githubusercontent.com/kadalu/kadalu/devel/manifests/kadalu-operator.yaml

File: extras/scripts/kadalu-quick-start-generic.sh [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        echo -e "Do you already have a disk allocation, if yes enter complete path of device or Create a new truncated file [path/create]"
+        read answer3
+        
+        if [ $answer3 == "path" ] ; then
+
+		echo -e "Enter the device file path"
+		read devPath
+
+		#example /home/vatsa/tryKadalu/disk/test-storage
+                sudo kubectl kadalu storage-add $storagePool --device $device:$devPath
```
Comment (amarts) [inline extras/scripts/kadalu-quick-start-generic.sh:36]: no need for kubectl kadalu... the script should run without python dependency!

File: extras/scripts/kadalu-quick-start-generic.sh [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        cp ../../examples/sample-app.yaml /tmp/${podFileName}.yaml
+	sed -i s/pod1/$podName/  /tmp/${podFileName}.yaml
+	sed -i s/pv1/$pvcName/  /tmp/${podFileName}.yaml
+	sudo kubectl create -f /tmp/$podFileName.yaml
+        
+        echo "Sleep 100s for status update"
+        sleep 100
+ 
+        #Status of the Pod
+        sudo kubectl get pods
```
Comment (amarts) [inline extras/scripts/kadalu-quick-start-generic.sh:100]: can you use `getopt` and take these variables as arguments instead. Also provide '--help'.

File: extras/scripts/kadalu-quick-start-generic.sh [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+	read podName
+
+
+        cp ../../examples/sample-app.yaml /tmp/${podFileName}.yaml
+	sed -i s/pod1/$podName/  /tmp/${podFileName}.yaml
+	sed -i s/pv1/$pvcName/  /tmp/${podFileName}.yaml
+	sudo kubectl create -f /tmp/$podFileName.yaml
+        
+        echo "Sleep 100s for status update"
+        sleep 100
```
Comment (amarts) [inline extras/scripts/kadalu-quick-start-generic.sh:97]: no need to sleep 100, but do it like https://github.com/kadalu/kadalu/blob/devel/tests/minikube.sh#L11

**Review by amarts — CHANGES_REQUESTED** [review]

**Comment by vatsa287** [comment]
Changes done through patch #327 .

**Pull request closed without merging**